pkgs/native_assets_builder/test/test_data/native_dynamic_linking_test.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@OnPlatform({
+  'mac-os': Timeout.factor(2),
+  'windows': Timeout.factor(10),
+})
+library;
+
+import 'dart:io';
+
+import 'package:native_assets_cli/native_assets_cli_internal.dart';
+import 'package:test/test.dart';
+
+import '../helpers.dart';
+
+void main() async {
+  late Uri tempUri;
+  const name = 'native_dynamic_linking';
+
+  setUp(() async {
+    tempUri = (await Directory.systemTemp.createTemp()).uri;
+  });
+
+  tearDown(() async {
+    await Directory.fromUri(tempUri).delete(recursive: true);
+  });
+
+  for (final dryRun in [true, false]) {
+    final testSuffix = dryRun ? ' dry_run' : '';
+    test('native_dynamic_linking build$testSuffix', () async {
+      final testTempUri = tempUri.resolve('test1/');
+      await Directory.fromUri(testTempUri).create();
+      final testPackageUri = testDataUri.resolve('$name/');
+      final dartUri = Uri.file(Platform.resolvedExecutable);
+
+      final processResult = await Process.run(
+        dartUri.toFilePath(),
+        [
+          'hook/build.dart',
+          '-Dout_dir=${tempUri.toFilePath()}',
+          '-Dpackage_name=$name',
+          '-Dpackage_root=${testPackageUri.toFilePath()}',
+          '-Dtarget_os=${OSImpl.current}',
+          '-Dversion=${HookConfigImpl.latestVersion}',
+          '-Dlink_mode_preference=dynamic',
```

Comment:
add `-Dlinking_enabled=0` (see fixes to tests in https://github.com/dart-lang/native/pull/1507)

Or possibly better, use `BuildConfig` constructor and write the config to a file and pass that in.

Reply:
Done